src/lib/charm/openstack/ovn_central.py

```diff
@@ -865,8 +867,20 @@ def assess_exporter(self):
             snap.refresh('prometheus-ovn-exporter', channel=channel)
         else:
             snap.install('prometheus-ovn-exporter', channel=channel)
+            fresh_install = True
         snap.connect_all()
 
+        # Note(mkalcok): After the plugs of the exporter snap are connected
+        # for the first time (on snap install), we need to restart
+        # the exporter service for the new permissions to take effect.
+        # The snap can also be installed by the snap layer, so we utilize
+        # additional flag to signal whether we already restarted the service.
+        not_initialized = not reactive.is_flag_set(
+            'prometheus-ovn-exporter.initialized')
+        if fresh_install or not_initialized:
+            ch_core.host.service_restart(self.exporter_service)
+            reactive.set_flag('prometheus-ovn-exporter.initialized')
```

Comment:
If the user removes the snap by clearing juju config `channel`, and then re-installs it by setting it again, this flag will be in a inconsistent state.

Do we need to clear the flag on `snap.remove()` above?

(On a related note, and not a problem of this patch, it appears to me that the remove by clearing `channel` feature would effectively have the [layer code](https://git.launchpad.net/layer-snap/tree/reactive/snap.py#n60) install and then the charm code remove the snap for every hook execution, but I don't expect you to address this now.)

Reply:
:+1: there's no harm on clearing the flag. It'll make the state more precise.